### PR TITLE
chore: remove hardcoded constants for ids

### PR DESCRIPTION
**Pull request by yquansah**
Closes #

Describe your proposed changes here.

Description of issue and what this PR seeks to achieve are detailed within this issue: https://github.com/influxdata/influxdb/issues/19547

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: kv/bucket.go [hunk 1]
```diff
@@ -182,7 +182,6 @@ func (s *Service) findBucketByName(ctx context.Context, tx Tx, orgID influxdb.ID
 		switch n {
 		case influxdb.TasksSystemBucketName:
 			return &influxdb.Bucket{
-				ID:              influxdb.TasksSystemBucketID,
 				Type:            influxdb.BucketTypeSystem,
 				Name:            influxdb.TasksSystemBucketName,
 				RetentionPeriod: influxdb.TasksSystemBucketRetention,
```
Comment (stuartcarnie) [inline kv/bucket.go:187]: This entire switch statement can be deleted and the `if IsNotFound(err)` statement only needs to return the `default` case:

```go
return nil, &influxdb.Error{
    Code: influxdb.ENotFound,
    Msg:  fmt.Sprintf("bucket %q not found", n),
}
```

File: kv/bucket.go [hunk 2]
```diff
@@ -361,7 +359,6 @@ func (s *Service) FindBuckets(ctx context.Context, filter influxdb.BucketFilter,
 
 	if needsSystemBuckets {
```
Comment (stuartcarnie) [inline kv/bucket.go:360]: Here is also legacy, so everything from:

```go
needsSystemBuckets := true
for _, b := range bs {
```

through to line 382 can be deleted.

File: tenant/storage_bucket.go [hunk 3]
```diff
@@ -120,7 +120,6 @@ func (s *Store) GetBucketByName(ctx context.Context, tx kv.Tx, orgID influxdb.ID
 		switch n {
```
Comment (stuartcarnie) [inline tenant/storage_bucket.go:120]: Same for this one, the `if kv.IsNotFound(err) {` need only be `return nil, ErrBucketNotFoundByName(n)` as the fixed system buckets are gone.

**Review by stuartcarnie — CHANGES_REQUESTED** [review]
Apologies if my issue was lacking some clarity, @yquansah! 

Previously, these system buckets did not exist in the key/value store, as we had hard coded them with IDs `10` and `11`. All the routines to read buckets had to manually materialize them. We have since made them real data in the key/value store, so these routines will fetch them as expected by reading the data from the key/value store.

**Comment by yquansah** [comment]
@stuartcarnie Excellent. Where does the logic live that stores these system buckets in key/value store? (Hopefully I am not misunderstanding)

**Comment by stuartcarnie** [comment]
Fair question, @yquansah. These are created automatically when an organization is created, and exist as real metadata in the key/value store:

https://github.com/influxdata/influxdb/blob/45a3f2e87c90c2e2d3fbeaae9a800aaa556c46ac/kv/bucket.go#L136-L156

We then look these buckets up by `name` to find them.

**Review by stuartcarnie — APPROVED** [review]
Great cleanup, @yquansah 🧹

**Pull request merged**